drivers/adc/Kconfig.mcux

```
@@ -37,11 +37,13 @@ config ADC_MCUX_LPADC
 	help
 	  Enable the MCUX LPADC driver.
 
+if ADC_MCUX_12B1MSPS_SAR || ADC_MCUX_LPADC
 config ADC_MCUX_ETC
 	bool "MCUX ADC ETC driver"
 	depends on HAS_MCUX_ADC_ETC
 	help
 	  Enable the MCUX ADC ETC driver.
+endif
```

```suggestion
config ADC_MCUX_ETC
	bool "MCUX ADC ETC driver"
	depends on DT_HAS_NXP_ADC_ETC_ENABLED
	help
	  Enable the MCUX ADC ETC driver.
```

and then just make a one-line binding file with the compat string, and add a small node to rt11xx.dtsi and rt10xx.dtsi that just looks like this basically:

```
adc_etc@deadbeef {
    compatible = "nxp,adc-etc";
};
```

can just leave status okay in soc level since the kconfig is `n` by default, if dont want to change all the board dts

---

why remove "if ADC_MCUX_12B1MSPS_SAR || ADC_MCUX_LPAD"?  ETC can not work standalone